ttnn/cpp/ttnn/operations/core/to_layout/to_layout_op.cpp

```diff
@@ -23,6 +23,18 @@ namespace core {
 
 namespace detail {
 
+inline bool nd_support(const ttnn::Tensor& tensor_arg, const ttnn::Layout layout) {
+    const auto initial_shape = tensor_arg.get_shape();
+    if (initial_shape.rank() > 4 && tensor_arg.get_layout() != layout) {
+        for (int i = 0; i < initial_shape.rank() - 4; i++) {
+            if (initial_shape[i] != 1) {
```

Comment:
This can be a TT_FATAL and you should print out i and initial_shape[i] in the message.